### PR TITLE
Add deployment pipeline files

### PR DESCRIPTION
**Pull request by francoislg**
[COM-765]

So this adds the Jenkinsfile to build and deploy the versions using the deployment pipeline.

It still needs Snyk, Veracode, and (potentially?) then auto-version-bump, but we at least have a basis.

[COM-765]: https://coveord.atlassian.net/browse/COM-765

### CODE REVIEW

File: .deployment.config.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
+    "product": "coveo-analytics-js",
+    "team_name": "commerce",
+    "general": {
+        "environments_order": {
+            "sequential": ["dev", "prd"]
```
Comment (toussaints) [inline .deployment.config.json:6]: is that what react-vapor does? dev to prod? Do we have a DNS in dev?

File: .deployment.config.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        },
+        "notifications": {
+            "slack_channels": ["commerceteambuilds"]
+        },
+        "team_jenkins": "commerceteambuilds",
+        "start_environment_automatically": true
```
Comment (toussaints) [inline .deployment.config.json:12]: not sure what that does

File: .deployment.config.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    },
+    "package_rollout": {
+        "only_consider_changesets_after": "688b6873442721d61e2a0ac492b9da808ddb0ef0"
```
Comment (toussaints) [inline .deployment.config.json:15]: TIL nice

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,61 @@
+#!groovy
+
+library(
+  identifier: 'jenkins-common-lib@v1.5',
```
Comment (toussaints) [inline Jenkinsfile:4]: I think the latest is 1.6

File: package.json [hunk 5]
```diff
@@ -12,9 +12,10 @@
         "build": "rollup -c",
         "start": "rollup -c -w --environment SERVE",
         "test": "jest --clearCache && jest --coverage",
+        "prepare-deploy": "mkdir -p deploy && cp dist/coveoua.js dist/coveoua.js.map deploy",
         "clean": "rimraf -rf dist dist_test coverage"
     },
-    "author": "Pierre-Alexandre St-Jean <pastjean@coveo.com>",
```
Comment (toussaints) [inline package.json:17]: RIP pastjean

File: read.version.sh [hunk 6]
```diff
@@ -1,5 +1,17 @@
...
+elif [ "$1" == "patch" ]
+then
+ echo `cat package.json | grep version | head -1 | awk -F: '{ print $2 }' | sed 's/[",]//g' | xargs`
+elif [ "$1" == "minor" ]
+then
+ echo `cat package.json | grep version | head -1 | awk -F: '{ print $2 }' | sed 's/[",]//g' | sed -E 's/\.[0-9]+$//g' | xargs`
+elif [ "$1" == "major" ]
+then
+ echo `cat package.json | grep version | head -1 | awk -F: '{ print $2 }' | sed 's/[",]//g' | sed -E 's/\.[0-9]+\.[0-9]+$//g' | xargs`
+fi
```
Comment (toussaints) [inline read.version.sh:17]: I guess you took that somewhere?

**Review by toussaints — APPROVED** [review]

File: .deployment.config.json (hunk 1 above)
Comment (francoislg) [reply .deployment.config.json:6]: react-vapor does QA to Prod. 

Not against putting QA here as well, but the terraform script in commerce-tools/coveoanalyticsjs is only executed in dev and prd, I felt it was more convenient.

We don't have a DNS in dev, put we could put one eventually :)

File: read.version.sh (hunk 6 above)
Comment (francoislg) [reply read.version.sh:17]: `read.version.sh` already existed with only the three `export`. I could not use those variables in Jenkins, so I adapted the script to be allowed to output it to stdout

If you have any better idea, let me know :smile:

**Review by cnguyen-coveo — APPROVED** [review]

File: .deployment.config.json (hunk 2 above)
Comment (francoislg) [reply .deployment.config.json:12]: not sure either

File: .deployment.config.json (hunk 1 above)
Comment (dlafreniere) [reply .deployment.config.json:6]: should we change the terraform to be `qa` and `prod` only?

**Review by dlafreniere — APPROVED** [review]

File: .deployment.config.json (hunk 1 above)
Comment (francoislg) [reply .deployment.config.json:6]: I don't know, `dev` => `prd` doesn't bother me :sweat_smile:

**Pull request merged**